web/src/lib/components/listen-box.tsx

```diff
@@ -138,7 +138,9 @@ export default class ListenBox extends Component<Props, State> {
           <Icon type="x"/>Nope.</a>
       </div>
       <audio className="audio-box"
-        src={this.props.src}
+        // Only include the src attribute if the source is defined
+        // (empty src attributes are invalid)
+        {...this.props.src && {src: this.props.src}}
```

Comment:
Thanks for the link! That explains the syntax :)

TBH, I don't like the compactness of this implementation in terms of readability though.

Have you considered pulling out the entire `<audio>` element into a local `AudioBox` component (name up to you, of course), where you can set a local variable based on the value of `src`, and then use the spread operator with that local variable, instead of combining it inside the interpolation syntax?

Similar to what's showing here: https://facebook.github.io/react/docs/conditional-rendering.html